logstash-core/lib/logstash/agent.rb

```diff
@@ -165,12 +165,15 @@ def collect_metrics?
     @collect_metric
   end
 
-  def create_pipeline(settings)
-    begin
-      config = fetch_config(settings)
-    rescue => e
-      @logger.error("failed to fetch pipeline configuration", :message => e.message)
-      return
+  def create_pipeline(settings, config=nil)
+
+    if config.nil?
+      begin
+        config = fetch_config(settings)
+      rescue => e
+        @logger.error("failed to fetch pipeline configuration", :message => e.message)
+        return
```

Comment:
it doesn't need, I think in this case the visual similarity between the two blocks helps reason about the behaviour of the code. but I can remove if ya want :)